src/main/java/codeu/controller/ForgetPasswordServlet.java

```diff
@@ -59,26 +49,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   /**
    * This function fires when user submits a form to reset password. It checks if the username is registered
    * and allows user to change password if it is. Otherwise, show an error to the user.
-   * // TODO : Add verification method. For now it just lets anyone reset any username's password.
+   * // TODO (My): Add verification method. For now it just lets anyone reset any username's password.
    * @param request
    * @param response
    * @throws IOException
    * @throws ServletException
    */
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
     throws IOException, ServletException {
     String username = request.getParameter("username");
-    String newPasswordHash = BCrypt.hashpw(request.getParameter("password"),BCrypt.gensalt());
+    String newPassword = request.getParameter("password");
 
     if(userStore.isUserRegistered(username)) {
       User user = userStore.getUser(username);
-      user.setPassword(newPasswordHash);
-      response.sendRedirect("/login");
+      user.setPassword(newPassword);
+      if(userStore.updateUserPassword(user, newPassword)) {
+        response.sendRedirect("/login");
+      } else {
```

Comment:
What if datastore fails to update the new password and things in this clause gets executed? New password would have been set to the user object in memory, but the user doesn't know because he/she gets the error message. Should we move line 66 user.setPassword() inside the if clause to make sure that datastore successfully update before updating the in-memory password?